### PR TITLE
support metadata, first pass

### PR DESCRIPTION
**Pull request by cmcarthur**

### CODE REVIEW

**Comment by nick-mccoy** [comment]
This looks mostly good to me -- the only revision I might have is to make sure fields that are used for bookmarking have their `inclusion` set to `automatic`.  It looks like this is the `Date` field for a subset of the streams:
https://github.com/fishtown-analytics/tap-campaign-monitor/blob/f56afd1cc9f12d654f81ab041d015cf644745597/tap_campaign_monitor/streams/base.py#L234-L237

**Comment by nick-mccoy** [comment]
Actually -- I see that `Date` is a key property for the relevant streams so it will be marked as `automatic` anyway.

File: tap_campaign_monitor/streams/base.py [hunk 1]
```diff
@@ -91,9 +115,17 @@ def sync_data(self, substreams=None):
...
-                    [self.filter_keys(obj)])
+                with singer.Transformer() as tx:
+                    singer.write_records(
+                        table,
+                        [tx.transform(
+                            obj,
+                            self.catalog.schema.to_dict(),
+                            singer.metadata.to_map(self.catalog.metadata))])
+
+                    singer.write_records(
```
Comment (KAllan357) [inline tap_campaign_monitor/streams/base.py]: Write records twice?

Comment (cmcarthur) [reply tap_campaign_monitor/streams/base.py]: thanks, caught & updated

**Pull request merged**